src/FX.Core/RedisStorageLayer.fs

```diff
@@ -5,9 +5,34 @@ open System
 
 open FsharpExchangeDotNetStandard
 
-open Newtonsoft.Json
+open System.Text.Json
+open System.Text.Json.Serialization
 open StackExchange.Redis
 
+[<AutoOpen>]
+module Serialization =
+    type DecimalTypeConverter() =
+        inherit JsonConverter<decimal>()
+
+        override this.Read(reader, _typeToConvert, _options) =
+            reader.GetDecimal()
+
+        override this.Write(writer, value, _options ) =
+            // Values such as 0m and 0.0m are equal, but have different string representations,
+            // which may cause problems when comparing serialized values.
+            // To avoid this, add 0.0m to value before writing, so all decimals will be serilized with fractional
+            // part, even if it's 0.
+            // For more info, see https://colinmackay.scot/2021/07/03/why-is-there-a-difference-between-decimal-0-and-0-0/
```

Comment:
@webwarrior-ws great comment and great investigation, however, let's improve some things here:
1. Instead of writing a converter, it's better to just make the tests be more liberal, and accept that x.0 is the same as x; let's write our own Assert.DecimalEquals function for example. We switched to FSharp.SystemTextJson to avoid converters!
2. This comment should go in the commit message as well

Reply:
If test was comparing decimals, there would be no problem. But test compares strings (serialized json).

Reply:
the tests are just making sure that Redis works; you could deserialize the JSON and make sure the ending object is the same (since they are records, an Equals invokation should work)